File: src/test/java/com/koddy/server/common/mock/fake/FakeEncryptor.java
package com.koddy.server.common.mock.fake;

import com.koddy.server.global.utils.encrypt.Encryptor;

public class FakeEncryptor implements Encryptor {
    private static final String DUMMY = "_koddy";

    @Override
    public String hash(final String value) {
        return value + DUMMY;
    }

    @Override
    public boolean matches(final String rawValue, final String encodedValue) {
        return encodedValue.replace(DUMMY, "").equals(rawValue);
    }

    @Override
    public String encrypt(final String value) {
        return value + DUMMY;
    }

    @Override
    public String decrypt(final String value) {
        return value.replace(DUMMY, "");
    }
}

File: src/test/java/com/koddy/server/common/mock/fake/FakeEncryptor.kt
package com.koddy.server.common.mock.fake

import com.koddy.server.global.utils.encrypt.Encryptor

class FakeEncryptor : Encryptor {
    override fun hash(value: String): String = value + UNI_DIRECTIONAL_DUMMY

    override fun matches(
        rawValue: String,
        encodedValue: String,
    ): Boolean = encodedValue.replace(UNI_DIRECTIONAL_DUMMY, "") == rawValue

    override fun encrypt(value: String): String = value + BI_DIRECTIONAL_DUMMY

    override fun decrypt(value: String): String = value.replace(BI_DIRECTIONAL_DUMMY, "")

    companion object {
        private const val UNI_DIRECTIONAL_DUMMY = "_hello_koddy_uni"
        private const val BI_DIRECTIONAL_DUMMY = "_hello_koddy_bi"
    }
}
